common/Jet_QA.C

```diff
@@ -385,98 +422,117 @@ void CommonJetQA(std::optional<uint32_t> trg = std::nullopt)
+          kinematicQA_Tracks -> setRestrictEtaRange(JetQA::RestrictEtaByR);
+          kinematicQA_Tracks -> setPtRange(ptJetRange.first, ptJetRange.second);
+          kinematicQA_Tracks -> setEtaRange(etaJetMaxRange.first, etaJetMaxRange.second);
+            if (trg.has_value())
+            {
+              kinematicQA_Tracks -> setTrgToSelect(trg.value());
+            }
+          se -> registerSubsystem(kinematicQA_Tracks);
+  	  }	  
 	}
-	se -> registerSubsystem(kinematicQA);
-	//this part is for no subtracted background towers
-	JetKinematicCheck* kinematicQA_NoSub = new JetKinematicCheck(
-                        "JetKinematicCheck" + trig_tag + "_tower_antikt",
-                        JetQA::JetInput[JetQA::Type::AntiKtTowerR02],
-                        JetQA::JetInput[JetQA::Type::AntiKtTowerR03],
-                        JetQA::JetInput[JetQA::Type::AntiKtTowerR04],
-                        JetQA::JetInput[JetQA::Type::AntiKtTowerR05]
-                        );
-        kinematicQA_NoSub -> Verbosity(verbosity);
-        kinematicQA_NoSub -> setHistTag("");
-        kinematicQA_NoSub -> setRestrictEtaRange(JetQA::RestrictEtaByR);
-        kinematicQA_NoSub -> setPtRange(ptJetRange.first, ptJetRange.second);
-        kinematicQA_NoSub -> setEtaRange(etaJetMaxRange.first, etaJetMaxRange.second);
-        if (trg.has_value())
-        {
-                kinematicQA_NoSub -> setTrgToSelect(trg.value());
-        }
-        se -> registerSubsystem(kinematicQA_NoSub);
-
-	//this part is for no subtracted background towers
-	
-        JetKinematicCheck* kinematicQA_Tracks = new JetKinematicCheck(
-                        "JetKinematicCheck" + trig_tag + "_track_antikt",
-                        JetQA::JetInput[JetQA::Type::AntiKtTrackR02],
-                        JetQA::JetInput[JetQA::Type::AntiKtTrackR03],
-                        JetQA::JetInput[JetQA::Type::AntiKtTrackR04],
-                        JetQA::JetInput[JetQA::Type::AntiKtTrackR05]
-                        );
-        kinematicQA_Tracks -> Verbosity(verbosity);
-        kinematicQA_Tracks -> setHistTag("");
-        kinematicQA_Tracks -> setRestrictEtaRange(JetQA::RestrictEtaByR);
-        kinematicQA_Tracks -> setPtRange(ptJetRange.first, ptJetRange.second);
-        kinematicQA_Tracks -> setEtaRange(etaJetMaxRange.first, etaJetMaxRange.second);
-        if (trg.has_value())
-        {
-                kinematicQA_Tracks -> setTrgToSelect(trg.value());
-        }
-        se -> registerSubsystem(kinematicQA_Tracks);	
 	
 	// create modules that take single R values ---------------------------------
 
+	std::vector<uint32_t> vecJetsToQA;
+   	if (JetQA::HasCalos)
+   	{
+     	  vecJetsToQA = JetQA::GetJetsToQA(JetQA::Source::Calos);
+     	  if (JetQA::HasTracks)
+          {
+            auto tmp = JetQA::GetJetsToQA(JetQA::Source::Tracks);
+            vecJetsToQA.insert(vecJetsToQA.end(), tmp.begin(), tmp.end());
+          }
+        }
```

Comment:
Careful! If you're running with only tracks, this block will never execute! Instead you could do something like so:
```suggestion
   	if (JetQA::HasCalos)
   	{
   	  auto vecCalJets = JetQA::GetJetsToQA(JetQA::Source::Calos);
   	  vecJetsToQA.insert(vecJetsToQA.end(), vecCalJets.begin(), vecCalJets.end());
        }
     	if (JetQA::HasTracks)
        {
          auto vecTrkJets = JetQA::GetJetsToQA(JetQA::Source::Tracks);
          vecJetsToQA.insert(vecJetsToQA.end(), vecTrkJets.begin(), vecTrkJets.end());
        }
```